packages/editor/src/components/block-list/block.js

```diff
@@ -372,6 +372,7 @@ export class BlockListBlock extends Component {
 			isPartOfMultiSelection,
 			isFirstMultiSelected,
 			isTypingWithinBlock,
+			isCaretWithinFormattedText,
```

Comment:
I don't like that the block's wrapper is aware of something very specific to the implementation of a given block (RichText). It doesn't seem like a generic enough solution for me. All blocks don't use RichText. As I see it the block wrapper should work similarily no matter what's the content of the `edit` function.

That said, I'm not familiar enough with the issue and the PR to propose an alternative. (which means I won't block this PR either)

Reply:
That's a good point. I don't have a better solution either off the top of my head, aside from just documenting that it's `RichText` specific, which still isn't great/is the whole problem. 🤷‍♂️